src/Datadog.Trace/Tracer.cs

```diff
@@ -421,7 +436,7 @@ private static string GetApplicationName()
             }
         }
 
-        private static IStatsd CreateDogStatsdClient(TracerSettings settings, string serviceName)
+        private static IStatsd CreateDogStatsdClient(TracerSettings settings, string serviceName, int? portOverride = null)
```

Comment:
Similar comment as [this one](https://github.com/DataDog/dd-trace-dotnet/pull/647/files#r381568366) in `Api`. The caller knows the value of `portOverride`, so it can replace the value of `TracerSettings.DogStatsdPort` before passing it in. This is not `Tracer`'s concern.

Reply:
The Tracer is the caller however. I can do the check in the delegate I subscribe?